test/Gen.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeApplications  #-}
+
+module Gen
+  ( mbb
+  , rtreeInt
+  ) where
+
+import Control.Applicative ((<$>))
+import Data.RTree.Base
+import Data.RTree.MBB      hiding (mbb)
+import Hedgehog
+import Prelude             hiding (length, lookup, map, null)
+import Text.Show.Functions ()
+
+import qualified Hedgehog.Gen   as G
+import qualified Hedgehog.Range as R
+
+mbb :: MonadGen m => m MBB
+mbb = G.shrink more $ do
+  cx  <- G.double (R.linearFrac (-100000.0) 100000.0)
+  cy  <- G.double (R.linearFrac (-100000.0) 100000.0)
+  h   <- G.double (R.linearFrac        0.0  100000.0)
+  w   <- G.double (R.linearFrac        0.0  100000.0)
+  return $ MBB (cx - w) (cy - h) (cx + w) (cy + h)
+  where more :: MBB -> [MBB]
+        more mbb'@(MBB ulx uly brx bry)
+            | isPointMBB mbb' = []
+            | otherwise       = [MBB (mid ulx brx) (mid uly bry) (mid ulx brx) (mid uly bry)]
+        mid x y = (y - x) / 2
```

Comment:
One benefit of `hedgehog` is you get automatic shrinking for free.  I supplied a custom shrinker here just in case you need that behaviour, but if you don't this can be dropped.